custom/enikshay/integrations/bets/views.py

```diff
@@ -0,0 +1,130 @@
+        self.status_code = status_code
+        super(ApiError, self).__init__(msg)
+
+
+class VoucherUpdate(jsonobject.JsonObject):
+    voucher_id = jsonobject.StringProperty(required=True)
+    payment_status = jsonobject.StringProperty(required=True, choices=['success', 'failure'])
+    payment_amount = jsonobject.IntegerProperty(required=False)
+    failure_description = jsonobject.StringProperty(required=False)
+
+    case_type = 'voucher'
+
+    @property
+    def case_id(self):
+        return self.voucher_id
+
+    @property
+    def properties(self):
+        if self.payment_status == 'success':
+            return {
+                'state': 'paid',
+                'amount_fulfilled': self.payment_amount,
+                'date_fulfilled': datetime.datetime.utcnow().date().isoformat(),
+            }
+        else:
+            return {
+                'state': 'rejected',
+                'reason_rejected': self.failure_description,
+                'date_rejected': datetime.datetime.utcnow().date().isoformat(),
+            }
+
+
+class IncentiveUpdate(jsonobject.JsonObject):
+    beneficiary_id = jsonobject.StringProperty(required=True)
+    episode_id = jsonobject.StringProperty(required=True)
+    payment_status = jsonobject.StringProperty(required=True, choices=['success', 'failure'])
+    payment_amount = jsonobject.IntegerProperty(required=False)
+    failure_description = jsonobject.StringProperty(required=False)
+    bets_parent_event_id = jsonobject.StringProperty(
+        required=False, choices=BETS_EVENT_IDS.values())
+
+    case_type = 'episode'
+
+    @property
+    def case_id(self):
+        return self.episode_id
+
+    @property
+    def properties(self):
+        status_key = 'tb_incentive_{}_status'.format(self.bets_parent_event_id)
+        if self.payment_status == 'success':
+            amount_key = 'tb_incentive_{}_amount'.format(self.bets_parent_event_id)
+            date_key = 'tb_incentive_{}_payment_date'.format(self.bets_parent_event_id)
+            return {
+                status_key: 'paid',
+                amount_key: self.payment_amount,
+                date_key: datetime.datetime.utcnow().date().isoformat(),
+            }
+        else:
+            date_key = 'tb_incentive_{}_rejection_date'.format(self.bets_parent_event_id)
+            reason_key = 'tb_incentive_{}_rejection_reason'.format(self.bets_parent_event_id)
+            return {
+                status_key: 'rejected',
+                date_key: datetime.datetime.utcnow().date().isoformat(),
+                reason_key: self.failure_description,
+            }
+
+
+def get_case(domain, case_id):
+    case_accessor = CaseAccessors(domain)
+    return case_accessor.get_case(case_id)
+
+
+def _update_case_from_request(request, domain, update_model):
+    try:
+        request_json = json.loads(request.body)
+    except ValueError:
+        return json_response({"error": "Malformed JSON"}, status_code=400)
```

Comment:
something we receive daily in "report style" that we can send to BETS and say "hey look, we got all these bad requests from you". Even just a separate log that doesn't have all the other requests in it could be a start.